### ISSUE TITLE
`Extractor:attributes:feature_level`: Introduce feature implementation levels

### ISSUE THREAD

**Issue by PeterKraus** [marda-alliance/metadata_extractors_schema#13]
This came up often in the past and was discussed at today's Office Hours: I think it would be great if we could implement several `feature_levels` for the `Extractor` schema. These should be strict supersets of each other. This is not the same as the version of the schema, which will allow implementing more features within a certain level (and deeper levels).

The current proposal is the following:

- `feature_level: 0` corresponds to an `Extractor` that can process a `FileType` and provides some sort of an output. The output format is completely unspecified at this level. This is intended to allow stepwise adoption of various extractors/parsers into this framework. For `v0.1` of this repo (#12) this is likely the only implemented `feature_level`.

Further proposed levels might be:

- `feature_level: 1` corresponds to an `Extractor` that processes a `FileType` and its output is in a "recognised" format, i.e. a `FileType` that can be opened and read (without data loss) using standard, open-source third-party tools (container formats such as `JSON`, `HDF5`, or `messagepack` come to mind).
- `feature_level: 2` corresponds to an `Extractor` implementation capable of providing information about the schema of the output file; whether this is `JSONschema` for a `JSON`, or any other relevant mechanism for specifying the schema of `netCDF` or `NeXus` files stored within `HDF5` containers.

@ml-evs What do you think?

**Comment by ml-evs** [comment]
I see how this would be useful! I think my remaining questions are:

- Are we expecting levels for 3/4/5 etc.? e.g., "can the extractor be executed using the common MaRDA API"?
- Can we come up with better meaningful names for these levels?
- and if so, can we just include them as `features: ["output", "generic", "specific", "common execution API"]`? ["output" would probably be omitted here as level 0 is a minimum...]

**Comment by PeterKraus** [comment]
Maybe `feature_level` is too generic, and it should be a nested structure such as `attributes:feature_level:output` or something like that. Then any future undefined key (such as `attributes:feature_level:api`) is, by convention, `level 0`. 

I like the integer values, as it makes comparisons easy: some downstream projects might be happy with any extraction into open formats (`level >= 1`) for archiving purposes, while some might require semantic understanding (say, `level >= 3`) for visualisation or postprocessing.

I think the obvious `feature_level: 3` in this chain would be semantic mapping (i.e., if there's an ontology mapping on what is "voltage", the schema should mark it as such). But I think we will know more about that once we are at `feature_level: 2`.

**Comment by ml-evs** [comment]
Another way of representing this maybe just hints at missing fields in this version of the spec:

```
schema: the output schema of this extractor (URL/URI whatever) --> implies level 2.
format: the default(?) output format of this extractor either a MaRDA file type (do we really want to register e.g. JSON, CSV etc...) or selection from an enum of common formats (messy) --> implies level 1
```

**Comment by PeterKraus** [comment]
I am slowly coming on board of the idea of having a set of optional `attributes`, which, if populated, signify a certain feature being available. The candidates for this treatment are:

- `installation`: the `Extractor` can be installed using the `api` following a prescribed syntax.
- `usage`: the `Extractor` can be executed using the `api` following a prescribed syntax. See #19.
- `supported_filetypes`: the `Extractor` ought to support at least one `Filetype`, see #18, however, if this entry is populated, the `Filetype` can be specified in the `usage` entry using a `{{ file_type }}` match.

Other features that we talked about can be something as follows:

- `output_format`: for `Extractors` that return a file (on disk) as opposed to an object (in memory), this describes the "container" format of that file, such as one of the "open" formats, such as `JSON`, `HDF5`, `NetCDF`, `NeXus`, or `csv` / `Ecsv`. In an ideal case, we'd be able to validate that the returned file is parseable using its standard library such as `json.load()` or `pandas.from_csv()`.
- `output_schema`: provides an URI/URL to the schema of the output data in a machine-actionable way.

**Comment by ml-evs** [comment]
Some agreement that most of this is solved for now by the `scope` field, added in #33
